pennylane/ops/op_math/linear_combination.py

```diff
@@ -0,0 +1,517 @@
+        return self
+
+    def eigvals(self):
+        """Return the eigenvalues of the specified operator.
+
+        This method uses pre-stored eigenvalues for standard observables where
+        possible and stores the corresponding eigenvectors from the eigendecomposition.
+
+        Returns:
+            array: array containing the eigenvalues of the operator
+        """
+        eigvals = []
+        for ops in self.overlapping_ops:
+            if len(ops) == 1:
+                eigvals.append(
+                    qml.utils.expand_vector(ops[0].eigvals(), list(ops[0].wires), list(self.wires))
+                )
+            else:
+                tmp_composite = Sum(*ops)  # only change compared to CompositeOp.eigvals()
+                eigvals.append(
+                    qml.utils.expand_vector(
+                        tmp_composite.eigendecomposition["eigval"],
+                        list(tmp_composite.wires),
+                        list(self.wires),
+                    )
+                )
+
+        return self._math_op(
+            qml.math.asarray(eigvals, like=qml.math.get_deep_interface(eigvals)), axis=0
+        )
+
+    def diagonalizing_gates(self):
+        r"""Sequence of gates that diagonalize the operator in the computational basis.
+
+        Given the eigendecomposition :math:`O = U \Sigma U^{\dagger}` where
+        :math:`\Sigma` is a diagonal matrix containing the eigenvalues,
+        the sequence of diagonalizing gates implements the unitary :math:`U^{\dagger}`.
+
+        The diagonalizing gates rotate the state into the eigenbasis
+        of the operator.
+
+        A ``DiagGatesUndefinedError`` is raised if no representation by decomposition is defined.
+
+        .. seealso:: :meth:`~.Operator.compute_diagonalizing_gates`.
+
+        Returns:
+            list[.Operator] or None: a list of operators
+        """
+        diag_gates = []
+        for ops in self.overlapping_ops:
+            if len(ops) == 1:
+                diag_gates.extend(ops[0].diagonalizing_gates())
+            else:
+                tmp_sum = Sum(*ops)  # only change compared to CompositeOp.diagonalizing_gates()
+                eigvecs = tmp_sum.eigendecomposition["eigvec"]
+                diag_gates.append(
+                    qml.QubitUnitary(
+                        qml.math.transpose(qml.math.conj(eigvecs)), wires=tmp_sum.wires
+                    )
+                )
+        return diag_gates
+
+    def map_wires(self, wire_map: dict):
+        """Returns a copy of the current ``LinearCombination`` with its wires changed according to the given
+        wire map.
+
+        Args:
+            wire_map (dict): dictionary containing the old wires as keys and the new wires as values
+
+        Returns:
+            .LinearCombination: new ``LinearCombination``
+        """
+        new_ops = tuple(op.map_wires(wire_map) for op in self.ops)
+        new_op = LinearCombination(self.data, new_ops)
```

Comment:
```suggestion
        coeffs, ops = self.terms()
        new_ops = tuple(op.map_wires(wire_map) for op in ops)
        new_op = LinearCombination(coeffs, new_ops)
```
data may contain extra things not in the coefficients.

Reply:
Nice catch, thanks!